e2e/scaffolding/extrinsicHelpers.ts

```diff
@@ -851,12 +856,38 @@ export class ExtrinsicHelper {
     );
   }
 
+  public static async waitForFinalization(ofBlockNumber?: number) {
+    const start = Date.now();
+    const blockNumber = ofBlockNumber || (await getBlockNumber());
+    let currentBlock = await getFinalizedBlockNumber();
+    while (currentBlock < blockNumber) {
+      if (start + 48_000 < Date.now()) {
+        throw new Error(
+          `Waiting for Finalized Block took longer than 48s. Waiting for "${blockNumber.toString()}", Current: "${currentBlock.toString()}"`
```

Comment:
This is a life saver. Some of these were taking forever and it was really hard to track down. Now they are capped.